<launch>
    <arg name="world_name" value="$(find edmo_sim)/worlds/empty_accelerated.world"/>
    <arg name="robot_1" value="edmo1" />
    <arg name="robot_2" value="edmo2" />
    <arg name="robot_3" value="edmo3" />
    <!-- <arg name="robot_4" value="edmo4" />
    <arg name="robot_5" value="edmo5" /> -->

    <group ns="$(arg robot_1)">
        <remap from="/clock" to="/$(arg robot_1)_clock"/>
        <env name="GAZEBO_MASTER_URI" value="http://localhost:11345"/>
        <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
        args=" -e ode $(arg world_name)" />

        <include file="$(find edmo_sim)/launch/init_snake.launch">
            <arg name="prefix" value="$(arg robot_1)" />
        </include>
    </group>

    <group ns="$(arg robot_2)">
        <remap from="/clock" to="/$(arg robot_2)_clock"/>
        <env name="GAZEBO_MASTER_URI" value="http://localhost:11346"/>
        <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
        args=" -e ode $(arg world_name)" />
        
        <include file="$(find edmo_sim)/launch/init_snake.launch">
            <arg name="prefix" value="$(arg robot_2)" />
        </include>
    </group>

    <group ns="$(arg robot_3)">
        <remap from="/clock" to="/$(arg robot_3)_clock"/>
        <env name="GAZEBO_MASTER_URI" value="http://localhost:11347"/>
        <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
        args=" -e ode $(arg world_name)" />

        <include file="$(find edmo_sim)/launch/init_snake.launch">
            <arg name="prefix" value="$(arg robot_3)" />
        </include>
    </group>

    <!-- <group ns="$(arg robot_4)">
        <remap from="/clock" to="/$(arg robot_4)_clock"/>
        <env name="GAZEBO_MASTER_URI" value="http://localhost:11348"/>
        <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
        args=" -e ode $(arg world_name)" />

        <include file="$(find edmo_sim)/launch/init_snake.launch">
            <arg name="prefix" value="$(arg robot_4)" />
        </include>
    </group>

    <group ns="$(arg robot_5)">
        <remap from="/clock" to="/$(arg robot_5)_clock"/>
        <env name="GAZEBO_MASTER_URI" value="http://localhost:11349"/>
        <node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen"
        args=" -e ode $(arg world_name)" />

        <include file="$(find edmo_sim)/launch/init_snake.launch">
            <arg name="prefix" value="$(arg robot_5)" />
        </include>
    </group> -->
</launch>